### PR TITLE
[13.0][FIX] sale_order_type: default sequence

### PR DESCRIPTION
**Pull request by chienandalu**
When the sale type has no sequence, the order gets the default one. When we change the type and the sequence is the same that that default, we should change the name, as we'd get a new reference of the same sequence.

cc @Tecnativa TT37832

please review @pedrobaeza @victoralmau

### CODE REVIEW

File: sale_order_type/models/sale.py [hunk 1]
```diff
@@ -33,6 +33,19 @@ def _default_type_id(self):
...
+    @api.model
+    def _default_sequence_id(self):
+        """We get the sequence in same way the core next_by_code method does so we can
+        get the proper default sequence"""
+        force_company = self.env.context.get("force_company")
+        if not force_company:
+            force_company = self.company_id.id or self.env.company.id
+        return self.env["ir.sequence"].search(
+            [("code", "=", "sale.order"), ("company_id", "in", [force_company, False])],
+            order="company_id",
```
Comment (cesar-tecnativa) [inline sale_order_type/models/sale.py:45]: why order="company_id" ?

File: sale_order_type/models/sale.py [hunk 2]
```diff
@@ -101,13 +114,25 @@ def create(self, vals):
     def write(self, vals):
         """A sale type could have a different order sequence, so we could
         need to change it accordingly"""
+        force_company = self.env.context.get("force_company")
+        if not force_company:
+            force_company = self.env.company.id
+        default_sequence = self._default_sequence_id()
```
Comment (cesar-tecnativa) [inline sale_order_type/models/sale.py:117]: Lines 117-119 force_company are included in function

File: sale_order_type/models/sale.py [hunk 3]
```diff
@@ -101,13 +114,25 @@ def create(self, vals):
...
         if vals.get("type_id"):
             sale_type = self.env["sale.order.type"].browse(vals["type_id"])
             if sale_type.sequence_id:
                 for record in self:
+                    # An order with a type without sequence would get the default one.
+                    # We want to avoid changing the order reference when the new
+                    # sequence has the same default sequence.
+                    ignore_default_sequence = (
+                        not record.type_id.sequence_id
+                        and sale_type.sequence_id == default_sequence
```
Comment (cesar-tecnativa) [inline sale_order_type/models/sale.py:127]: should be sale_type.sequence_id.id == default_sequence.id?

**Review by cesar-tecnativa — CHANGES_REQUESTED** [review]

**Review by victoralmau — APPROVED** [review]

File: sale_order_type/models/sale.py (hunk 3 above)
Comment (chienandalu) [reply sale_order_type/models/sale.py:127]: We don't need to. We're comparing at object level.

File: sale_order_type/models/sale.py (hunk 2 above)
Comment (chienandalu) [reply sale_order_type/models/sale.py:117]: Mmm, yes. You're right

File: sale_order_type/models/sale.py (hunk 1 above)
Comment (chienandalu) [reply sale_order_type/models/sale.py:45]: Is the same order `next_by_code()` does it: https://github.com/OCA/OCB/blob/13.0/odoo/addons/base/models/ir_sequence.py#L280

**Review by cesar-tecnativa — APPROVED** [review]

**Comment by pedrobaeza** [comment]
/ocabot merge patch

Please fw-port it to upper versions

**Pull request merged**